python/lsst/meas/algorithms/ingestIndexReferenceTask.py

```diff
@@ -104,15 +109,16 @@ class DatasetConfig(pexConfig.Config):
     ref_dataset_name = pexConfig.Field(
         dtype=str,
         default='cal_ref_cat',
-        doc='String to pass to the butler to retrieve persisted files.',
+        doc="Name of this reference catalog to be used in the butler registry.",
+        deprecated="When gen2 butler is removed, this field will become non-optional, with no default."
```

Comment:
Yeah, I've been torn on how to handle this one. We'll need people to override it (to completely get rid of `cal_ref_cat`!), but if we remove the default right now, that would actually break some gen2 stuff. The deprecation warning here is rather annoying; I could create a ticket blocked on the gen2 removal ticket (DM-20572) to remove this default during or after that process, since it really only matters in gen3 for refcat creation, which is something that very few users actually do.

Reply:
My issue here is that the deprecation warning will come when people use this _correctly_ (setting an override) and there won't be a warning when people don't override it (which is when we'd need a warning).  So the deprecation code that we have just doesn't work.  I think we just need to file a ticket with a TODO here.